### ISSUE TITLE
:state/.state= broken in edit_line and edit_box

### ISSUE THREAD

**Issue by lmrodriguezr** [shoes/shoes4#1183]
Hi all,

It looks like the `:state` argument to `edit_line` and `edit_box` is being silently ignored, and the `.state=` method works but disables selection (regardless of the value passed). This little app demonstrates the problem:
```ruby
Shoes.app do
   stack margin_left: 40, margin_right: 20  do
      %w(readonly disabled).each do |s|
         title s
         edit_line("Always enabled", state: s)
         edit_line(:state => s).text = "Always enabled"
         edit_line("Always disabled").state = s
         para ""
      end
      title "This really should be enabled:"
      edit_line("but is not!").state="enabled"
      edit_line("but is not!").state=:enabled
      edit_line("but is not!").state=true
      edit_line("This one is!").state=nil
   end
end
```

The first two methods `edit_line("txt", state: s)` and `edit_line(state: s)` produce regular editable `edit_line`s. The third method `.state=s` seems to ignore the value passed (except if `s.nil?`) and produces greyed-out and un-selectable boxes, which kind of defeats [the original purpose](http://comments.gmane.org/gmane.comp.lib.shoes/2152) of these methods.

In case the problem is specific to my system: I'm running shoes4 out of the box (7fc6c6c5685a6a04ef3d677289c3525e0ad46719) with jruby 9.0.3.0 in Mac 10.10.5.

Thanks!
--Miguel.

**Comment by PragTob** [comment]
Hey there,

thank you for the great bug report with a nice app to reproduce it!

Sorry about that, might be related to one of our style related changes, it should be rather easy to debug I hope :)

Tobi

**Comment by PragTob** [comment]
Official [docs](http://shoesrb.com/manual/Styles.html#:state):

>The :state style is for disabling or locking certain controls, if you don't want them to be edited.
>
>Here are the possible style settings:
>
>    nil - the control is active and editable.
>    "readonly" - the control is active but cannot be edited.
>    "disabled" - the control is not active (grayed out) and cannot be edited.

**Comment by PragTob** [comment]
Hi there,

this should be fixed over in #1186 !

It should now (after it lands on master) only be disabled for "disabled"/:disabled and it also works when passing them right in.

I'll make an issue for the readonly style

Thanks again for the bug report!

**Comment by PragTob** [comment]
That was merged and fixed, will be in pre5 whenw e release it
